service/src/main/kotlin/io/provenance/explorer/grpc/v1/ExchangeGrpcClient.kt

```diff
@@ -0,0 +1,81 @@
+package io.provenance.explorer.grpc.v1
+
+import cosmos.base.query.v1beta1.Pagination
+import io.grpc.ManagedChannelBuilder
+import io.provenance.exchange.v1.AccountAmount
+import io.provenance.exchange.v1.Market
+import io.provenance.exchange.v1.MarketBrief
+import io.provenance.exchange.v1.QueryGetAllMarketsRequest
+import io.provenance.exchange.v1.QueryGetMarketCommitmentsRequest
+import io.provenance.exchange.v1.QueryGetMarketRequest
+import io.provenance.exchange.v1.QueryGrpcKt
+import io.provenance.explorer.config.interceptor.GrpcLoggingInterceptor
+import org.springframework.stereotype.Component
+import java.net.URI
+import java.util.concurrent.TimeUnit
+
+@Component
+class ExchangeGrpcClient(channelUri: URI) {
+
+    private val exchangeClient: QueryGrpcKt.QueryCoroutineStub
+
+    init {
+        val channel =
+            ManagedChannelBuilder.forAddress(channelUri.host, channelUri.port)
+                .also {
+                    if (channelUri.scheme == "grpcs") {
+                        it.useTransportSecurity()
+                    } else {
+                        it.usePlaintext()
+                    }
+                }
+                .idleTimeout(5, TimeUnit.MINUTES)
+                .keepAliveTime(60, TimeUnit.SECONDS)
+                .keepAliveTimeout(20, TimeUnit.SECONDS)
+                .intercept(GrpcLoggingInterceptor())
+                .build()
+
+        exchangeClient = QueryGrpcKt.QueryCoroutineStub(channel)
+    }
+
+    suspend fun getMarket(marketId: Int): Market = exchangeClient.getMarket(
+        QueryGetMarketRequest.newBuilder().apply {
+            this.marketId = marketId
+        }.build()
+    ).market
+
+    suspend fun getMarketBriefsByDenom(denom: String): List<MarketBrief> = exchangeClient.getAllMarkets(
```

Comment:
not sure how we typically order function, but I'm a fan of least protected to most protected, or pub to priv. Your org is fine to me, just curious what our standard should be going forward, or if we even want one.